playbooks/fundamentals/template_development_guide.adoc

```diff
@@ -0,0 +1,387 @@
+---
+---
+= OpenShift Templates Development Guide
+Eric Sauer <esauer@redhat.com>
+:toc: macro
+:toc-title:
+
+This doc provides some guidance, tips, and best practices for using, developing and managing templates in OpenShift.
+
+toc::[]
+
+== Overview
+
+link:https://docs.openshift.com/container-platform/3.7/dev_guide/templates.html[OpenShift Templates] are best known as the way in which the OpenShift Web Console is populated with quickstart applications and other content. However, they are also a very powerful tool that, used thoughtfully, can be the building block of an _Infrastructure as Code_ solution for managing many aspects of cluster and application state.
+
+In this guide, we dive deep into the usage and makeup of OpenShift templates, covering first how they can be used, then how to write custom templates, and finally, how to put all of this knowledge together to create powerful automation workflows for managing workloads in OpenShift.
+
+== OpenShift Template Discovery
+
+We're going to start by doing an interactive, exploratory deep into templates, starting with the basics and working our way deeper.
+
+=== Kickin' it off with some oc new-app
+
+In order to get kicked off in template exploration, let's start by spinning up a Java application using `oc new-app`.
+
+```
+$ oc new-app --template=openjdk18-web-basic-s2i
+--> Deploying template "openshift/openjdk18-web-basic-s2i" to project eric-test
+
+     Red Hat OpenJDK 8
+     ---------
+     Application template for Java applications built using S2I.
+
+     A new java application has been created in your project.
+
+     * With parameters:
+        * Application Name=openjdk-app
+        * Custom http Route Hostname=
+        * Git Repository URL=https://github.com/jboss-openshift/openshift-quickstarts
+        * Git Reference=master
+        * Context Directory=undertow-servlet
+        * Github Webhook Secret=4sPr4Br3 # generated
+        * Generic Webhook Secret=V218WEHy # generated
+        * ImageStream Namespace=openshift
+
+--> Creating resources ...
+    service "openjdk-app" created
+    route "openjdk-app" created
+    imagestream "openjdk-app" created
+    buildconfig "openjdk-app" created
+    deploymentconfig "openjdk-app" created
+--> Success
+    Build scheduled, use 'oc logs -f bc/openjdk-app' to track its progress.
+    Run 'oc status' to view your app.
+```
+
+Looking at what was done here, we used an `oc` command to instantiate a template by name. The template we used (`openjdk18-web-basic-s2i`) is a template that has been preloaded into OpenShift for convenience. We can also see that several objects were created as a result, including a `Service`, `Route`, `ImageStream`, `BuildConfig` and a `DeploymentConfig`. We can take a look at these templates using some oc commands.
```

Comment:
In addition to calling out `oc` command, do you want to introduce oc = OpenShift Command Line tool?